index.d.ts

```diff
@@ -44,10 +44,7 @@ declare namespace Moleculer {
 		trace(...args: any[]): void;
 	}
 
-	type ActionHandler<T = any, S = ServiceSettingSchema> = (
-		this: Service<S>,
-		ctx: Context<any, any>
-	) => Promise<T> | T;
+	type ActionHandler<T = any> = (ctx: Context<any, any>) => Promise<T> | T;
```

Comment:
why remove this:service ?

Reply:
I believe it's because `this` is now set by the `ThisType<Service<s>>` on the entire Action schema on line 476 and similarly for other types that make up the schema.

Reply:
> I believe it's because `this` is now set by the `ThisType<Service<s>>` on the entire Action schema on line 476 and similarly for other types that make up the schema.

That is correct.  The `ThisType` utilization is intended to solve the original problem that adding the `this:Service` was intended to solve, without introducing side effects when you attempt to use an `ActionHandler` (or other type) in a context where `this` isn't `Service`.

It is important to note (as indicated in the original PR comments) that the `noImplicitThis` compiler option must be enabled for `ThisType` to function properly.